src/cli/credentials.h

```diff
@@ -30,22 +34,18 @@ class Basic_Credentials_Manager : public Botan::Credentials_Manager
    {
    public:
       Basic_Credentials_Manager(bool use_system_store,
-                                const std::string& ca_paths)
+                                const std::string& ca_path)
          {
-         std::vector<std::string> paths;
-
-         if(ca_paths.empty() == false)
-            paths.push_back(ca_paths);
-
-         if(use_system_store)
+         if(ca_path.empty() == false)
             {
-            paths.push_back("/etc/ssl/certs");
-            paths.push_back("/usr/share/ca-certificates");
+            std::shared_ptr<Botan::Certificate_Store> cs(new Botan::Certificate_Store_In_Memory(ca_path));
+            m_certstores.push_back(cs);
             }
 
-         if(paths.empty() == false)
+         if(use_system_store)
             {
-            load_certstores(paths);
+            std::shared_ptr<Botan::Certificate_Store> cs(new Botan::System_Certificate_Store);
+            m_certstores.push_back(cs);
```

Comment:
Why not:

```suggestion
            m_certstores.push_back(std::make_shared<Botan::Certificate_Store>());
```

Reply:
GCC 4.8 has a bug converting unique_ptr<U> to unique_ptr<T> when U derives T. I'm not sure if shared_ptr has the same problem, I just reflexively worked around it. I'll check if it is a problem for shared_ptr as well.

Reply:
shared_ptr works, unique_ptr doesn't.